e2e/testcafe-devextreme/tests/dataGrid/keyboardNavigation/keyboardNavigation.functional.ts

```diff
@@ -4729,3 +4729,56 @@ test('DataGrid - Data rows are skipped during Tab navigation if the first column
   showBorders: true,
   width: 300,
 }));
+
+test('DataGrid input cell should not put tabindex to incorrect element while on edit mode (T1239462)', async (t) => {
+  const dataGrid = new DataGrid('#container');
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+  const button = Selector('.dx-datagrid-addrow-button');
+
+  await t
+    .click(button)
+    .pressKey('tab');
+  await takeScreenshot('data-grid_keyboard-navigation-input-text-focused.png', dataGrid.element);
+
+  await t.expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await createWidget('dxDataGrid', {
+    showBorders: true,
```

Comment:
We should use the simplest configuration in the test. Moreover, it is very helpful if you make the project simple in the reproducing stage.

```
  showBorders: true,
    selection: {
      mode: 'multiple',
    },
    editing: {
      mode: 'form',
      allowAdding: true,
    },
    columns: [
      'Calculation',
      'CalculationType',
      {
        type: 'buttons',
        fixed: true,
      },
    ],
    dataSource: {
      store: [],
    },
```

Reply:
Got it :) Thanks